src/main/java/teammates/storage/api/FeedbackResponsesDb.java

```diff
@@ -938,7 +978,9 @@ private FeedbackResponse getFeedbackResponseEntity(
         List<FeedbackResponse> firstQueryResponses =
             (List<FeedbackResponse>) q.execute(feedbackQuestionId, receiver, section);
         for(FeedbackResponse response : firstQueryResponses){
-            FeedbackResponseList.put(response.getId(), response);
+            if (!JDOHelper.isDeleted(response)) {
+                FeedbackResponseList.put(response.getId(), response);
```

Comment:
You get to rename this variable too :D. Let's not call it a *List here actually ...